src/registers/debug.rs

```diff
@@ -0,0 +1,580 @@
+
+    fn try_from(size: usize) -> Result<Self, Self::Error> {
+        Self::new(size).ok_or(TryFromIntError(()))
+    }
+}
+
+/// A valid value of the [`Dr7`] debug register.
+///
+/// In addition to the [`Dr7Flags`] this value has a condition field and a size field for each debug address register.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[repr(transparent)]
+pub struct Dr7Value {
+    bits: u64,
+}
+
+impl From<Dr7Flags> for Dr7Value {
+    fn from(dr7_flags: Dr7Flags) -> Self {
+        Self::from_bits_truncate(dr7_flags.bits())
+    }
+}
+
+impl Dr7Value {
+    const fn valid_bits() -> u64 {
+        let field_valid_bits = (1 << 32) - (1 << 16);
+        let flag_valid_bits = Dr7Flags::all().bits();
+        field_valid_bits | flag_valid_bits
+    }
+
+    /// Convert from underlying bit representation, unless that representation contains bits that do not correspond to a field.
+    #[inline]
+    pub const fn from_bits(bits: u64) -> Option<Self> {
+        if (bits & !Self::valid_bits()) == 0 {
+            Some(Self { bits })
+        } else {
+            None
+        }
+    }
+
+    /// Convert from underlying bit representation, dropping any bits that do not correspond to fields.
+    #[inline]
+    pub const fn from_bits_truncate(bits: u64) -> Self {
+        Self {
+            bits: bits & Self::valid_bits(),
+        }
+    }
+
+    /// Convert from underlying bit representation, preserving all bits (even those not corresponding to a defined field).
+    ///
+    /// # Safety
+    ///
+    /// The bit representation must be a valid [`Dr7Value`].
+    #[inline]
+    pub const unsafe fn from_bits_unchecked(bits: u64) -> Self {
+        Self { bits }
+    }
+
+    /// Returns the raw value of the fields currently stored.
+    #[inline]
+    pub const fn bits(&self) -> u64 {
+        self.bits
+    }
+
+    /// Returns the [`Dr7Flags`] in this value.
+    #[inline]
+    pub const fn flags(self) -> Dr7Flags {
+        Dr7Flags::from_bits_truncate(self.bits)
+    }
+
+    /// Inserts the specified [`Dr7Flags`] in-place.
+    #[inline]
+    pub fn insert_flags(&mut self, flags: Dr7Flags) {
+        self.bits |= flags.bits();
+    }
+
+    /// Removes the specified [`Dr7Flags`] in-place.
+    #[inline]
+    pub fn remove_flags(&mut self, flags: Dr7Flags) {
+        self.bits &= !flags.bits();
+    }
+
+    /// Toggles the specified [`Dr7Flags`] in-place.
+    #[inline]
+    pub fn toggle_flags(&mut self, flags: Dr7Flags) {
+        self.bits ^= flags.bits();
+    }
+
+    /// Inserts or removes the specified [`Dr7Flags`] depending on the passed value.
+    #[inline]
+    pub fn set_flags(&mut self, flags: Dr7Flags, value: bool) {
+        if value {
+            self.insert_flags(flags);
+        } else {
+            self.remove_flags(flags);
+        }
+    }
+
+    /// Returns the condition field of a debug address register.
+    pub fn get_condition(&self, n: DebugAddressRegisterNumber) -> HwBreakpointCondition {
```

Comment:
Getters typically don't have a `get_` prefix, so this should just be called `condition`.